crates/subspace-networking/src/node_runner.rs

```diff
@@ -657,7 +658,39 @@ where
 
                 kademlia.remove_peer(&peer_id);
             }
+
+            self.add_observed_external_address(info.observed_addr);
+        }
+    }
+
+    fn add_observed_external_address(&mut self, observed_addr: Multiaddr) {
+        // TODO: replace with Autonat
+        // TODO: won't work with QUIC
```

Comment:
Maybe link to PRs where this feature was removed by default and explains why Autonat is necessary and why QUIC is not working. Just some additional context, please.